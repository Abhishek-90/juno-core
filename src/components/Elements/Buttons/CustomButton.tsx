/* eslint-disable react/button-has-type */
import * as React from 'react'
import styled from 'styled-components'
import * as theme from '../../../constants/themeConstants'

interface ICustomButton {
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void
  className?: string
  type?: 'submit' | 'reset' | 'button'
  disabled?: boolean
  icon?: {}
  label: string
  suppressed?: boolean
}

interface IButton {
  suppressed: boolean | undefined
}

const Button = styled.button<IButton>`
  display: inline-block;
  font-weight: 400;
  color: ${(props) =>
    props.suppressed ? `${theme.colorGrey} ` : `${theme.colorBlack} `};
  text-align: center;
  vertical-align: middle;
  user-select: none;
  background-color: transparent;
  border: 1px solid transparent;
  border-top-color: transparent;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-left-color: transparent;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
  font-family: 'Raleway Variable', sans-serif;

  &:hover {
    color: ${theme.colorBlack};
  }

  &:disabled {
    cursor: not-allowed;
  }
`

const InnerButton = styled.div`
  display: flex;
  align-items: center;

  .icon {
    margin-right: 13px;
    line-height: 0;
    text-align: center;
    transition: opacity 0.3s ease 0s;
  }
`

const CustomButton = (props: ICustomButton) => {
  const { onClick, className, disabled, icon, label, type, suppressed } = props
  return (
    <Button
      onClick={onClick ? (event) => onClick(event) : undefined}
      className={className}
      type={type ?? 'button'}
      disabled={disabled}
      suppressed={suppressed}
    >
      <InnerButton>
        {icon && <div className="icon">{icon}</div>}
        <span>{label}</span>
      </InnerButton>
    </Button>
  )
}
export default CustomButton

CustomButton.defaultProps = {
  onClick: undefined,
  type: 'button',
  disabled: false,
  icon: null,
  className: null,
  suppressed: false,
}
