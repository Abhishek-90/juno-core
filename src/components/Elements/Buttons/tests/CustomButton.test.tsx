/* eslint-disable no-console */
import renderer from 'react-test-renderer'
import toJson from '../../../../utils/toJSONforTest'
import CustomButton from '../CustomButton'

test('Button renders with text if label is provided', () => {
  const component = renderer.create(
    <CustomButton label="Test button" title="test" />
  )
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Button fires the onClick event when clicked', () => {
  const component = renderer.create(
    <CustomButton
      label="Test button"
      onClick={() => {
        console.log('testing')
      }}
      title="test"
    />
  )
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()

  // manually trigger the callback
  tree.props.onClick()

  // re-rendering
  tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
