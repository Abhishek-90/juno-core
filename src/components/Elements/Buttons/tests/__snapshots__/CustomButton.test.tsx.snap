// Vitest Snapshot v1

exports[`Button fires the onClick event when clicked 1`] = `
<button
  className="sc-bczRLJ bBRcZQ"
  disabled={false}
  onClick={[Function]}
  title="test"
  type="button"
>
  <div
    className="sc-gsnTZi PBjap"
  >
    <span>
      Test button
    </span>
  </div>
</button>
`;

exports[`Button fires the onClick event when clicked 2`] = `
<button
  className="sc-bczRLJ bBRcZQ"
  disabled={false}
  onClick={[Function]}
  title="test"
  type="button"
>
  <div
    className="sc-gsnTZi PBjap"
  >
    <span>
      Test button
    </span>
  </div>
</button>
`;

exports[`Button renders with text if label is provided 1`] = `
<button
  className="sc-bczRLJ bBRcZQ"
  disabled={false}
  title="test"
  type="button"
>
  <div
    className="sc-gsnTZi PBjap"
  >
    <span>
      Test button
    </span>
  </div>
</button>
`;
